File: ios/ios/di/PokemonDatabaseProviderFactory.swift
//
//  PokemonDatabaseProviderFactory.swift
//  ios
//
//  Created by Harshith Shetty on 03/05/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import Foundation
import shared

class PokemonDatabaseProviderFactory: IDatabaseDriverFactory {
    func createDriver() -> SqlDriver {
        return NativeSqliteDriver.init(schema: PokemonDatabaseCompanion.init().Schema, name: "pokemon.db", maxReaderConnections: 4)
    }

}

File: iosUmbrellaModule/src/iosMain/kotlin/com/justadeveloper96/pokedex_kmp/iosUmbrellaModule/database/PokemonDatabaseProviderFactory.kt
package com.justadeveloper96.pokedex_kmp.iosUmbrellaModule.database

import com.justadeveloper96.pokedex_kmp.feature_pokemon_list.PokemonDatabase
import com.justadeveloper96.pokedex_kmp.helpers.dao.IDatabaseDriverFactory
import com.squareup.sqldelight.db.SqlDriver
import com.squareup.sqldelight.drivers.native.NativeSqliteDriver

class PokemonDatabaseProviderFactory : IDatabaseDriverFactory {
    override fun createDriver(): SqlDriver {
        return NativeSqliteDriver(
            schema = PokemonDatabase.Schema,
            name = "pokemon.db",
            maxReaderConnections = 4
        )
    }
}
